schemes/musica/tuvx/musica_ccpp_tuvx_height_grid.F90

```diff
@@ -0,0 +1,173 @@
+  ! ************************ (exo values) *****************************
+  ! ------(top)------ i_int = 1           -------(top)------ i_int = pver + 2
+  !                                       ================== i_mid = pver + 1
+  ! ================= i_mid = 1           ------------------ i_int = pver + 1
+  ! ----------------- i_int = 2           ================== i_mid = pver
+  !                                       ------------------ i_int = pver
+  !        ||
+  !        ||                                     ||
+  !                                               ||
+  ! ----------------- i_int = pver
+  ! ================= i_imd = pver        ------------------ i_int = 2
+  !                                       ================== i_mid = 1
+  ! -----(ground)---- i_int = pver+1      -----(ground)----- i_int = 1
+  !
+  !-----------------------------------------------------------------------
+
+  !> Label for height grid in TUV-x
+  character(len=*), parameter, public :: height_grid_label = "height"
+  !> Units for height grid in TUV-x
+  character(len=*), parameter, public :: height_grid_units = "km"
+
+contains
+
+  !> Creates a TUVX height grid from the host-model height grid
+  function create_height_grid( vertical_layer_dimension, &
+      vertical_interface_dimension, errmsg, errcode ) result( height_grid )
+
+    use musica_ccpp_util, only: has_error_occurred
+    use musica_tuvx_grid, only: grid_t
+    use musica_util,      only: error_t
+
+    integer,          intent(in)  :: vertical_layer_dimension     ! (count)
+    integer,          intent(in)  :: vertical_interface_dimension ! (count)
+    character(len=*), intent(out) :: errmsg
+    integer,          intent(out) :: errcode
+    type(grid_t),     pointer     :: height_grid
+
+    type(error_t) :: error
+
+    height_grid => null()
+    if ( vertical_layer_dimension < 1 ) then
+      errmsg = "[MUSICA Error] Invalid vertical_layer_dimension."
+      errcode = 1
+      return
+    end if
+    if ( vertical_interface_dimension - vertical_layer_dimension /= 1 ) then
+      errmsg = "[MUSICA Error] Invalid vertical_interface_dimension."
+      errcode = 1
+      return
+    end if
+    height_grid => grid_t( height_grid_label, height_grid_units, &
+                           vertical_interface_dimension, error )
+    if ( has_error_occurred( error, errmsg, errcode ) ) return
+
+  end function create_height_grid
+
+  !> Sets TUVX height grid values from the host-model height grid
+  subroutine set_height_grid_values( height_grid, host_midpoints, &
+      host_edges, errmsg, errcode )
+
+    use ccpp_kinds,       only: kind_phys
+    use musica_ccpp_util, only: has_error_occurred
+    use musica_tuvx_grid, only: grid_t
+    use musica_util,      only: error_t
+
+    type(grid_t),     intent(inout) :: height_grid
+    real(kind_phys),  intent(in)    :: host_midpoints(:) ! km
+    real(kind_phys),  intent(in)    :: host_edges(:)     ! km
```

Comment:
Is the term edge equivalent to interface? If so, can we use interface across the code base to enhance the consistency and readability?
```suggestion
    real(kind_phys),  intent(in)    :: host_interface(:)     ! km
```